src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -109,41 +109,53 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
         /// <param name="rootScriptPath">The root function directory.</param>
         public void PurgeOldFiles(string rootScriptPath)
         {
-            if (!Directory.Exists(rootScriptPath))
+            try
             {
-                return;
-            }
+                if (!Directory.Exists(rootScriptPath))
+                {
+                    return;
+                }
 
-            // Create a lookup of all potential functions (whether they're valid or not)
-            // It is important that we determine functions based on the presence of a folder,
-            // not whether we've identified a valid function from that folder. This ensures
-            // that we don't delete logs/secrets for functions that transition into/out of
-            // invalid unparsable states.
-            var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
+                // Create a lookup of all potential functions (whether they're valid or not)
+                // It is important that we determine functions based on the presence of a folder,
+                // not whether we've identified a valid function from that folder. This ensures
+                // that we don't delete logs/secrets for functions that transition into/out of
+                // invalid unparsable states.
+                var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
 
-            var secretsDirectory = new DirectoryInfo(_secretsPath);
-            foreach (var secretFile in secretsDirectory.GetFiles("*.json"))
-            {
-                if (string.Compare(secretFile.Name, ScriptConstants.HostMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0)
+                var secretsDirectory = new DirectoryInfo(_secretsPath);
+                if (!Directory.Exists(_secretsPath))
```

Comment:
I added this not exists check, and the rest of the changes are me pushing the entire method into a try/catch